### ISSUE TITLE
[RFE] Support for additional authentication 

### ISSUE THREAD

**Issue by aelbarkani** [redhat-cop/vault-config-operator#43]
Thanks for this awesome operator!

I would like to suggest adding extra authentication methods in addition to Kubernetes one. Maybe using a Vault token can be enough in a first step (like here in cert-manager: https://cert-manager.io/docs/configuration/vault/#authenticating-with-a-token)

**Comment by raffaelespazzoli** [comment]
If I get it correctly, you mean additional methods to authenticate to Vault, not additional auth engines. What would the use case be?
the kubernetes method is designed to use the pod identity to authenticate to Vault.

**Comment by aelbarkani** [comment]
Yes exactly. We use [SPIFFE](https://spiffe.io/)  to authenticate to Vault, either through JWT or through TLS client certificates (including Kubernetes workloads) by using an identity federator which gives identities to workloads (something similar to what is done here: https://cloud.redhat.com/blog/fine-grained-iam-roles-for-openshift-applications). The reason for that is the difficulty to manage the Kubernetes authentication in Vault at scale : if you have dozens of Kubernetes clusters to manage and also dozens of Vault instances, it means that every time you add cluster you need to change Vault configuration in order to add the additional cluster. That can be OK if the team who manages the Vault instances is the same as the one who manages the Kubernetes clusters. But if they are different it can be a mess. And it's even a bigger mess if you have several teams managing several fleets of Kubernetes and several fleets of Vault instances.

**Comment by aelbarkani** [comment]
P.S: the reason why I proposed Vault token authentication instead of OIDC/JWT or TLS is that Vault token is more flexible, since it's more universal/flexible and leaves the the choice of getting Vault token to the user by using any authentication method he wants.

**Comment by raffaelespazzoli** [comment]
LDAP authentication has been added.
